src/components/ui/Table/Table.tsx

```diff
@@ -0,0 +1,185 @@
+/* eslint-disable react/jsx-key */
+import { memo, useEffect } from 'react';
+import type {ReactNode } from 'react'
+import { useTable, usePagination } from 'react-table';
+import type { TableHeaderProps, HeaderPropGetter, Column } from 'react-table'
+
+import classnames from 'classnames';
+
+import { DEFAULT_PAGE_SIZE } from '../../../constants';
+
+import Pagination from './components/Pagination';
+import TokenPagination from './components/TokenPagination';
+import type { PaginationOptions, TokenPaginationOptions } from './constants';
+import type { PaginationControl } from './components/Pagination';
+
+import styles from './Table.module.scss';
+
+// REF https://github.com/TanStack/react-table/discussions/2664
+type ColumnOptions = {
```

Comment:
Tried that but it just won't work, and the technical debt for the declaration's typing is huge. Rewriting it or extending it with interface is not an option, so we can only assert it now

Reply:
I see `ColumnInterface` here - https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/react-table
Can we use it and extend it? What was the issue when you tried?

Reply:
The problem is that `ColumnInterface` is an internal type, used by `ColumnInstace`
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/207f8177bfad6b542e5f1cb9b1936af6fb280816/types/react-table/index.d.ts#L107-L110

As with extends we can only extend the interface by not override it, we have no problem add in new fields but we cannot change the type interface for existing one declared in the interface. Some types there do not take and expose the generic correctly, and it is nested 3 - 5 levels deep and merge with some other interfaces again.

I haven't stash or commit when my type is not correct, I can't reproduce it to you right now, but simply using extends in `global.d.ts` cannot solve the issue

Reply:
Ok, let's add (update) more types then (types are from react-table docs) https://react-table.tanstack.com/docs/api/useTable

`Cell?: Function | React.Component => JSX`
`Header: String | Function | React.Component => JSX`
`accessor: String`

For accessor, I guess it's enough to have `String` type since I doubt we'll use it as a function

I am afraid there will be errors for you on MAA when you will use this
E.g. `Cell` has an argument that won't be typed

Please check that https://gist.github.com/Grsmto/1ac3a7ddb8ad8288660784a37bff1798
I think it can be useful

Reply:
Just to confirm do you mean adding types to this interface?

https://github.com/LBHackney-IT/mtfh-finance-components/blob/4c5c8a9851bd51be424b774148e3d6301632d93a/src/components/ui/Table/Table.tsx#L19-L29

Reply:
Yes

Reply:
@vadim-kirpun I have just added that and there should be no type error now